cmd/uplink/cmd/mount.go

```diff
@@ -185,6 +197,26 @@ func (sf *storjFS) Mkdir(name string, mode uint32, context *fuse.Context) fuse.S
 	return fuse.OK
 }
 
+func (sf *storjFS) Rmdir(name string, context *fuse.Context) (code fuse.Status) {
+	zap.S().Debug("Rmdir: ", name)
+
+	objects, err := sf.listObjects(sf.ctx, name, true)
```

Comment:
It looks wasteful to first collect all objects to be deleted in a slice and then iterate the slice to delete them one by one. If the folder being deleted contains thousands of object, this can consume quite some memory and will be slower.

Can't you call `sf.store.List` inline here and process each page of the listing as it comes?

Reply:
Almost the same struct is in OpenDir and I wanted to avoid duplication. I will do this in a different way.